icloudpd/email_notifications.py

```diff
@@ -12,14 +12,17 @@ def send_2sa_notification(
 ):
     """Send an email notification when 2SA is expired"""
     to_addr = to_addr if to_addr else smtp_email
+    from_addr = smtp_email if smtp_email else to_addr
     logger = setup_logger()
     logger.info("Sending 'two-step expired' notification via email...")
     smtp = smtplib.SMTP()
     smtp.set_debuglevel(0)
     smtp.connect(smtp_host, smtp_port)
     if not smtp_no_tls:
         smtp.starttls()
-    smtp.login(smtp_email, smtp_password)
+
+    if smtp_email is not None or smtp_password is not None:
```

Comment:
I think this should be `and` instead of `or`?

Reply:
Maybe. It was a conscious choice on my part, because I was thinking that maybe you had an SMTP-server that required a username but the password was blank. Or maybe an SMTP-server without any username requirement, and only a password requirement.

But I am happy to change it to `and`, if you feel that it makes more sense.

Reply:
Oh right, I just wasn't sure if those cases were possible with SMTP. But yes the `or` makes sense, we may as well support those cases as well!